extensions/cornerstone/src/tools/ImageOverlayViewerTool.tsx

```diff
@@ -0,0 +1,307 @@
+    const overlayTopLeftWorldPos = utilities.imageToWorldCoords(imageId, [
+      x - 1, // Remind that top-left corner's (x, y) is be (1, 1)
+      y - 1,
+    ]);
+    const overlayTopLeftOnCanvas = viewport.worldToCanvas(
+      overlayTopLeftWorldPos
+    );
+    const overlayBottomRightWorldPos = utilities.imageToWorldCoords(imageId, [
+      width,
+      height,
+    ]);
+    const overlayBottomRightOnCanvas = viewport.worldToCanvas(
+      overlayBottomRightWorldPos
+    );
+
+    // add image to the annotations svg layer
+    const svgns = 'http://www.w3.org/2000/svg';
+    const svgNodeHash = `image-overlay-${_id}`;
+    const existingImageElement = svgDrawingHelper.getSvgNode(svgNodeHash);
+
+    const attributes = {
+      'data-id': svgNodeHash,
+      width: overlayBottomRightOnCanvas[0] - overlayTopLeftOnCanvas[0],
+      height: overlayBottomRightOnCanvas[1] - overlayTopLeftOnCanvas[1],
+      x: overlayTopLeftOnCanvas[0],
+      y: overlayTopLeftOnCanvas[1],
+      href: overlayData.dataUrl,
+    };
+
+    if (
+      isNaN(attributes.x) ||
+      isNaN(attributes.y) ||
+      isNaN(attributes.width) ||
+      isNaN(attributes.height)
+    ) {
+      console.warn(
+        'Invalid rendering attribute for image overlay',
+        attributes['data-id']
+      );
+      return false;
+    }
+
+    if (existingImageElement) {
+      _setAttributesIfNecessary(attributes, existingImageElement);
+      svgDrawingHelper.setNodeTouched(svgNodeHash);
+    } else {
+      const newImageElement = document.createElementNS(svgns, 'image');
+      _setNewAttributesIfValid(attributes, newImageElement);
+      svgDrawingHelper.appendNode(newImageElement, svgNodeHash);
+    }
+    return true;
+  }
+
+  /**
+   * Get viewport's referene image id,
+   * TODO: maybe we should add this to the BaseTool or AnnotationTool class
+   * as it is often used by tools.
+   *
+   * @param viewport
+   * @returns
+   */
+  private _getReferencedImageId(
+    viewport: Types.IStackViewport | Types.IVolumeViewport
+  ): string {
+    const targetId = this.getTargetId(viewport);
+
+    let referencedImageId;
+
+    if (viewport instanceof StackViewport) {
+      referencedImageId = targetId.split('imageId:')[1];
+    }
+
+    return referencedImageId;
+  }
+}
+
+function _setAttributesIfNecessary(attributes, svgNode: SVGElement) {
+  Object.keys(attributes).forEach(key => {
+    const currentValue = svgNode.getAttribute(key);
+    const newValue = attributes[key];
+    if (newValue === undefined || newValue === '') {
+      svgNode.removeAttribute(key);
+    } else if (currentValue !== newValue) {
+      svgNode.setAttribute(key, newValue);
+    }
+  });
+}
+
+function _setNewAttributesIfValid(attributes, svgNode: SVGElement) {
+  Object.keys(attributes).forEach(key => {
+    const newValue = attributes[key];
+    if (newValue !== undefined && newValue !== '') {
+      svgNode.setAttribute(key, newValue);
+    }
+  });
+}
```

Comment:
These are already implemented in the cs3d drawingSVG, can we import it from there?